src/Http/Http.Abstractions/src/Extensions/UsePathBaseExtensions.cs

```diff
@@ -26,7 +26,7 @@ public static IApplicationBuilder UsePathBase(this IApplicationBuilder app, Path
             }
 
             // Strip trailing slashes
-            pathBase = pathBase.Value?.TrimEnd('/');
+            pathBase = pathBase.Value?.TrimEnd('/')!;
```

Comment:
This one doesn't look right. It's using this operator:

https://github.com/dotnet/aspnetcore/blob/4e01bbe4cfeb8d63cf5ac40fcf1a6a7c77c7e911/src/Http/Http.Abstractions/src/PathString.cs#L460-L461

Reply:
Oh, that's a bug. Value is already un-escaped, and ConvertFromString would un-escape it again. Double un-escaping is problematic in some contexts. This should be:
```suggestion
            pathBase = new PathString(pathBase.Value?.TrimEnd('/'));
```